### ISSUE TITLE
Before running autorouter kicad users shall be asked to run Checkers

### ISSUE THREAD

**Issue by dpalic** [freerouting/freerouting#320]
### Please provide as many details as you can
* What version of freerouting are you using?
1.9.0

* On what platform did you run it? Windows, Linux, MacOS? 32-bit or 64-bit?
Linux, but also Windows (both 64bit)

* What EDA do you use? KiCad, KiCad with freerouting plugin, EAGLE, EasyEDA, other? What is its exact version?
KiCad 8.0.3

* What types of files are you working with? Can you attach the project files (.pro, .kicad_pcb, etc) or the exported DSN file? (Please note that GitHub doesn't allow DSN file uploads, so you need to ZIP it before uploading.)
schematics and pcb

* Did you use any command line arguments?
no

* Can you post a screenshot of the problem?
no

* Can you include the latest, related log entries from the .log file?
no

* Please list the detailed steps to reproduce the problem
do a usual process. Schematics, afterwards PCB, run freeroute.
In my case, in the meanwhile a bunch of persons happend the issue.
They started the freeroute and it took multiple hours. Until they identified that the missed some errors on the schematics and the PCB.
A easy way would be to ask the user on a modal dialog if they want to like this

```
+----------------------------------------------------------------------------------------+
| The autorouting can take multiple hours, depending on the complexity of your PCB.      |
| It is highly recommended to run the design rules checkers on Schematics and PCB        | 
| before starting autorouting.                                                           |
|                                                                                        |
| +--------------------------------------------------+                                   |
| |execute the design rules checker on Schematics now|                                   |
| +--------------------------------------------------+                                   |
|                                                                                        |
| +-------------------------------------------+                                          |
| |execute the design rules checker on PCB now|                                          |
| +-------------------------------------------+                                          |
|                                                                                        |
| +---------------------------------------------------+                                  |
| |both checks already done, please start autorouting |                                  |
| +---------------------------------------------------+                                  |
|----------------------------------------------------------------------------------------+
```
With such a simple UX friendly change the users can easily get guided through the freeroute to face a much better user experience instead of running into an issue after many hours autorouting to identify, they forgot to do some basic checks.

If you name us where we could change such a behaviour with some examples, we would give it a try to implement it.

* What would you expect freerouting to do in this situation?

to show the named modal dialog and even to call the checker

* Is fixing this issue critical for your work?

if you count hours of blocked time also critical, I would say yes

**Comment by andrasfuchs** [comment]
I agree and I like the idea. Would you try to implement this [in our KiCad integration script](https://github.com/freerouting/freerouting/blob/master/integrations/KiCad/kicad-freerouting/plugins/plugin.py)?
